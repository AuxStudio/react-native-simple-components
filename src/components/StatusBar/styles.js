import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    zIndex: 100,
  },
  statusBar: {
    height: 22, // ios status bar height
  },
});

export default styles;
