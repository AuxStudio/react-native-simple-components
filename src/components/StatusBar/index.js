import React from 'react';
import PropTypes from 'prop-types';
import { View, StatusBar, Platform } from 'react-native';

import styles from './styles';
import styleConstants from '../../styleConstants';

const propTypes = {
  backgroundColor: PropTypes.string,
  barStyle: PropTypes.string,
};

const defaultProps = {
  backgroundColor: styleConstants.colors.white,
  barStyle: 'dark-content',
};

const StatusBarComponent = ({ backgroundColor, barStyle }) => {
  const iosStatusBar = Platform.OS === 'ios' && (
    <View style={[styles.statusBar, { backgroundColor }]} />
  );

  return (
    <View style={[styles.container]}>
      <StatusBar backgroundColor={backgroundColor} barStyle={barStyle} />
      {iosStatusBar}
    </View>
  );
};

StatusBarComponent.propTypes = propTypes;
StatusBarComponent.defaultProps = defaultProps;

export default StatusBarComponent;
