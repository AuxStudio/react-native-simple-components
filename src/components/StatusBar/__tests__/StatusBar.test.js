import React from 'react';
import renderer from 'react-test-renderer';

import StatusBar from '../';

describe('StatusBar', () => {
  it('renders with all props', () => {
    expect(
      renderer.create(<StatusBar backgroundColor="red" barStyle="dark-content" />),
    ).toMatchSnapshot();
  });

  it('renders with minimum required props', () => {
    expect(renderer.create(<StatusBar />)).toMatchSnapshot();
  });
});
