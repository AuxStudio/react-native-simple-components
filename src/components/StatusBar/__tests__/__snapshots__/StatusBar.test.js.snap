// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusBar renders with all props 1`] = `
<View
  style={
    Array [
      Object {
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 100,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "height": 22,
        },
        Object {
          "backgroundColor": "red",
        },
      ]
    }
  />
</View>
`;

exports[`StatusBar renders with minimum required props 1`] = `
<View
  style={
    Array [
      Object {
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 100,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "height": 22,
        },
        Object {
          "backgroundColor": "#FFFFFF",
        },
      ]
    }
  />
</View>
`;
